[ansible-test] Add Fedora 33

relrod opened the pull request:
##### SUMMARY

Change:
- Changes to make F33 work with current tests
- Add F33 to docker.txt
- Add F33 to AZP config

Test Plan:
- ci_complete

Signed-off-by: Rick Elrod <rick@elrod.me>

<!--- Describe the change below, including rationale and design decisions -->

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->

- Test Pull Request

##### COMPONENT NAME

ansible-test

webknjaz commented on test/lib/ansible_test/_data/completion/docker.txt:
```diff
@@ -7,6 +7,7 @@ centos8 name=quay.io/ansible/centos8-test-container:1.21.0 python=3.6 seccomp=un
 fedora30 name=quay.io/ansible/fedora30-test-container:1.17.0 python=3.7
 fedora31 name=quay.io/ansible/fedora31-test-container:1.17.0 python=3.7
 fedora32 name=quay.io/ansible/fedora32-test-container:1.17.0 python=3.8
+fedora33 name=quay.io/ansible/fedora33-test-container:1.24.0 python=3.9
```
Is it possible to also make use of the 3.10 alpha that's available in F33? There seems to be demand for that (https://github.com/ansible/ansible/pull/72852).

mattclay replied:
Support for 3.10 is something that should be worked on separately from this PR.

webknjaz replied:
Totally, I just wanted to raise attention to that PR.

bek24 commented:
Looks wrong !!

mattclay commented on .azure-pipelines/azure-pipelines.yml:
```diff
@@ -122,6 +122,8 @@ stages:
             - name: Fedora 32
               test: fedora32
+            - name: Fedora 33
+              test: fedora33
```
We only keep the latest two Fedora releases, so we should drop Fedora 31 now.

mattclay commented on test/lib/ansible_test/_data/requirements/constraints.txt:
```diff
@@ -58,7 +58,7 @@ isort == 4.3.15
 mccabe == 0.6.1
 pylint == 2.3.1
-typed-ast == 1.4.0  # 1.4.0 is required to compile on Python 3.8
+typed-ast == 1.4.1  # 1.4.1 is required to compile on Python 3.9
```
I'd make the changes for pylint in a separate PR, since they're likely going to be backported differently than the other changes in this PR.

relrod replied:
I can do it in a separate commit in the same PR, but doing a separate PR will be hard since they're needed for tests to pass ;)

mattclay replied:
I'd like to see the pylint changes in a separate PR that we merge first, then re-run tests on this PR before merging.

webknjaz committed:
Declare support for Python 3.9 in the metadata

Co-authored-by: Rick Elrod <rick@elrod.me>

relrod committed:
[ansible-test] Add Fedora 33 support and drop F31

Change:
- Changes to make F33 work with current tests
- Add F33 to docker.txt
- Add F33 to AZP config

Test Plan:
- ci_complete

Signed-off-by: Rick Elrod <rick@elrod.me>

relrod commented:
# NOTE:

If someone merges this before I do -- please keep the commits intact, don't squash them.

mattclay commented:
This is currently blocked on a separate PR (https://github.com/ansible/ansible/pull/72972) which avoids pylint test errors on Python 3.9.

mattclay commented on changelogs/fragments/ansible-test-fedora33.yml:
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - ansible-test - Add support for running tests on Fedora 33 (https://github.com/ansible/ansible/pull/72861).
+  - ansible-test - Drop support for the Fedora 31 container (https://github.com/ansible/ansible/pull/72861).
```
Although we're removing it from CI, it's not being removed from `ansible-test`:

```suggestion
```

mattclay commented:
/azp run
